### DISCUSSION TITLE
2 devices, second is not recognized

### DISCUSSION THREAD

**Discussion by Sill1000** [HomeSpan/HomeSpan#527] [Q&A] [answered]
Hello,

I have a bridge that already uses the ID 46637726. And I would like to add a second device.
I took a second ESP that I programmed with example 3. 
To be able to pair it without disturbing the bridge, I found another QR code (84131633).
 So from the serial monitor:
S 84131633 then restart.
 Scan QR code, add device to "Apple Home" and see "Connecting to the switch". After a while "Is you device connected to the Wifi ?" and finally " Accessory not found. "
But I can ping the ESP. 

Shoud I change the S-code  but how to calculate it  ? What is wrong ? 

Thanks

**Comment by HomeSpan** [comment, marked as answer]
See this [FAQ](https://github.com/HomeSpan/HomeSpan/blob/master/docs/FAQ.md#can-you-use-more-than-one-homespan-device-on-the-same-homekit-network) for using multiple devices.  Also, note that the QR code is NOT the same as the pairing code.  Typing 'S 84131633" changes the pairing code on a device.  This is the code you type into Apple Home App when pairing.  If you want to create a QR code that you can use to work with that pairing code, see the [QR Code](https://github.com/HomeSpan/HomeSpan/blob/master/docs/QRCodes.md) page for details.  Note that the 4-character Setup ID (which is not the same as a the 8-digit pairing code) on each device must be unique.

**Reply by Sill1000** [reply]
sorry it's not clear for me... :-( 
According to FAQ:

> The only thing you need to ensure is that you assign a different Display Name to each of your devices. 

On my 1st bridge I don't have any Ceiling Fan.

homespan.begin(Category:: Fans, "Ceiling Fan");  
After:
S 84131633
Tried to pair it with this code with the keyboard but always: "accessory not found"

with discovery _hap_tcp : Ceiling fan is present

**Comment by HomeSpan** [comment]
Just to confirm:

1) You have two ESP32's running HomeSpan at the same time.  Each has a different display name (the second argument) in `homeSpan.begin()`
2) One of the devices is paired and working with HomeKit.
3) You are not able to pair the second device.
4) You have verified that the second device is connected to your WiFi network.

Try resetting the pairing parameters on the second device by typing 'H' into the Serial Monitor.  This will erase any previous pairing information as well as generate a new random Device ID.  When the device reboots it should say it indicate that it is unpaired.

There is no reason to re-issue the 'S' command.  Once you change the pairing code, it is retained until reset, or the device is FULLY erased.  Typing 'H' does not erase the pairing code.

Also, note that the 8-digit numeric pairing code DOES NOT have to be unique.  You can use the same pairing code (or the default code that comes with HomeSpan) for every device without any issue.

**Reply by Sill1000** [reply]
thanks  for your  help !

> 1. You have two ESP32's running HomeSpan at the same time.  Each has a different display name (the second argument) in `homeSpan.begin()`

1: `homeSpan.begin(Category::Bridges, "Bridge");`
2: `homeSpan.begin(Category:: Fans, "Ceiling Fan");`

> 2. One of the devices is paired and working with HomeKit.

Right !
> 3. You are not able to pair the second device.

Right !
> 4. You have verified that the second device is connected to your WiFi network.

Right and I can "ping" it. 

> 
> Try resetting the pairing parameters on the second device by typing 'H' into the Serial Monitor. This will erase any previous pairing information as well as generate a new random Device ID. When the device reboots it should say it indicate that it is unpaired.
Tried "H"; "Please pair with homekit" on the serial monitor. 
On the App by the keyboard: 46637726
 
Not possible to add the accessory. Operation aborted.
BUT... but...
by scanning the "standard 46637726" QR code it works ! (on the serial monitor: *** Error: Unrecognized Controller Pairing ID") Why this error is comming now ?
Why only the QR is ok ?
But it works !! :-)



> There is no reason to re-issue the 'S' command. Once you change the pairing code, it is retained until reset, or the device is FULLY erased. Typing 'H' does not erase the pairing code.
> 
> Also, note that the 8-digit numeric pairing code DOES NOT have to be unique. You can use the same pairing code (or the default code that comes with HomeSpan) for every device without any issue.

**Nice information.** I thought that we canot use again the same code on 2 devices ! 

To resume changing with the pairing code  84131633 is not ok. by keyboard entry or by scanning the QR code 
[https://www.npmjs.com/package/homekit-qrcode](url) 
Why ??

But everything is ok with standard code.

**Comment by HomeSpan** [comment]
I just wanted to check back on this.  Are you still finding that you cannot change the pairing code?

To double-check, you should be doing the following:

1) First, ensure the device is NOT paired to HomeKit.  You can force this from the CLI by typing 'H'.
2) Change the setup code using the CLI by typing 'S 33445566'.  This changes the setup code to 33445566, but does not result in a reboot.  However, a hash of code is saved so that it will be remembered even if the device is rebooted.
3) Check that your sketch does NOT include anything like this: `homeSpan.setPairingCode("46637726");` since that will reset the setup code every time the sketch runs.
4) On your iPhone, open the Home App and click the + button in the upper right.
5) Select _Add Accesory_
6) From the screen that opens, select _More options..._ at the top
7) You should see an icon for your device - click it.  If you don't see the icon, something is wrong -- the device is either not able to connect to your wifi network, your router is not passing through MDNS messages, or the device thinks it is already paired (in which case the Home App will not display it's icon).
8) Once you select the icon, a new menu appears telling you that device is not certified.  Click _Add Anyway_.
9) Now type 33445566 as the setup code.  In the most recent versions of the Home App, this will appear as 3344-5566.
10) The Home App should begin the process to pair.  If the setup code is incorrect, it will explicitly tell you the code is wrong.  You should not get a device not found warning, since you would not have been able to make it past step 7 if the device was not found.

Note the above procedure has nothing to do with QR codes.  You cannot use a QR code to pair unless you prepare the correct QR code as well as set a separate ID in the sketch.

**Comment by Sill1000** [comment]
> 7. You should see an icon for your device - click it.  If you don't see the icon, something is wrong -- the device is either not able to connect to your wifi network, your router is not passing through MDNS messages, or the device thinks it is already paired (in which case the Home App will not display it's icon).

Icon is present!
> 8. Once you select the icon, a new menu appears telling you that device is not certified.  Click _Add Anyway_.


Problem found ! 
With the iPhone this menu is covered by the keyboard. you can just touch the "cancel" button... and you can write your code 3344-5566 and get "Not possible to add the accessory"!!!!!
Tried with my iPad and select add anyway and it works !!! 

The behavior of Apple device's is a bit strange... with "Cancel" it should cancel the pairing and leave all menus!


> 9. Now type 33445566 as the setup code.  In the most recent versions of the Home App, this will appear as 3344-5566.

Right !



> Note the above procedure has nothing to do with QR codes. You cannot use a QR code to pair unless you prepare the correct QR code as well as set a separate ID in the sketch.

This QR code is working: [https://github.com/HomeSpan/HomeSpan/blob/master/docs/QRCodes.md](url)

But this QR code is not working ? What to do if I would like to use it ? S84131633 is not enough because it enable only the "writed" code.
![Homekit 84131633](https://user-images.githubusercontent.com/88202590/226734345-298a5c5e-c7e5-4c79-841c-381964290be7.jpg)

**Reply by HomeSpan** [reply]
Please review all of the HomeSpan documentation on QR codes.  It explains that when you make a QR code, the code embeds not only the 8-digit pairing code, but also a 4-character alphanumeric code.  You must set that code in the sketch, and then you need to generate a QR code that includes it.  I recommend using Apple's HomeKit Simulator to generate QR codes, or use the characters output by HomeSpan whenever you change the pairing code.  These characters can be used by any generic QR generator.  As noted, all the details are provided in the documentation.
